### PR TITLE
Feature/kls 684 export academy signin

### PR DESCRIPTION
**Pull request by davidu1975**
New Sign in page for Aventri users with a Great account migrating to Great Export Academy from Aventri

**Context**
This PR adds the first version of the export academy signin flow. In this version, the user is already registered for the export academy via Aventri and also has a Great account, and has received a unique link where their email address will be pre-populated and they just need to sign in

The backend implementation of the unique link has not yet been created, so the view currently relies on a registration uuid in the query parameters of the URL. Once the backend has been implemented this will be updated accordingly.

**Reusing IOO code**
This implementation reuses the logic used by international online offer's login view which uses django, rather than the main login flow using react. Currently

Some IOOLogin code has been moved to sso_mixins

**To test**
- Make sure you have an EA registration for your chosen email in your local db
- Make sure you have directory-forms-api, directory-sso and directory-sso-proxy running locally
- Go to `export-academy/signin?registration-id=<your_registration_id>`
- Your email address should be pre-populated
- You should be able to go through the signin

Screenshot

Screenshot 2023-06-22 at 15 29 23

### Workflow

- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/KLS-684
- [x] Jira ticket has the correct status.
- [x] A clear/description pull request messaged added.

### Merging

- [ ] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

File: export_academy/views.py [hunk 1]
```diff
@@ -463,3 +463,44 @@ def do_validate_code_flow(self, request):
...
+        initial = super().get_initial()
+        user = Registration.objects.get(pk=self.request.GET.get('registration-id'))
+        initial['email'] = user.email
+        return initial
+
+    def get_login_url(self):
+        return self.request.build_absolute_uri(reverse('core:login'))
+
+    def do_sign_in_flow(self, request):
+        form = forms.SignInForm(request.POST)
```
Comment (mforner13) [inline export_academy/views.py:483]: I realised when doing my current ticket, you can just use `self.get_form()` to get the form related to a formview which is a bit nicer :)

File: sso/mixins.py [hunk 2]
```diff
@@ -56,3 +57,55 @@ def handle_verification_code_success(self, upstream_response, redirect_url):
...
+            uidb64 = verification_code.pop('user_uidb64')
+            token = verification_code.pop('verification_token')
+            sso_helpers.send_verification_code_email(
+                email=email,
+                verification_code=verification_code,
+                form_url=request.path,
+                verification_link=self.get_verification_link(uidb64, token),
+                resend_verification_link=self.get_resend_verification_link(),
+            )
+            return 'Email unverified: we have re-sent you an email containing a link to verify your email address'
```
Comment (mforner13) [inline sso/mixins.py:96]: Out of interest - is this the same error that appears when the same scenario happens on normal great login?

**Review by mforner13 — COMMENTED** [review]
Couple of comments :)

File: export_academy/templates/export_academy/includes/upgrade_body_text.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<p>We’ve upgraded the UK Export Academy website and it has now joined all our other services on great.gov.uk.</p>
```
[the remark this answers is not in the document]
Comment (davidu1975) [reply export_academy/templates/export_academy/includes/upgrade_body_text.html:1]: done

File: export_academy/templates/export_academy/includes/upgrade_form_fields.html [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{% csrf_token %}
```
[the remark this answers is not in the document]
Comment (davidu1975) [reply export_academy/templates/export_academy/includes/upgrade_form_fields.html:1]: done

[the remark this answers is not in the document]
Comment (davidu1975) [reply export_academy/templates/export_academy/includes/upgrade_form_fields.html:1]: done

File: export_academy/views.py [hunk 5]
```diff
@@ -463,3 +463,44 @@ def do_validate_code_flow(self, request):
...
+    success_url = reverse_lazy('export_academy:upcoming-events')
+
+    def get_initial(self):
+        initial = super().get_initial()
+        user = Registration.objects.get(pk=self.request.GET.get('registration-id'))
+        initial['email'] = user.email
+        return initial
+
+    def get_login_url(self):
+        return self.request.build_absolute_uri(reverse('core:login'))
```
[the remark this answers is not in the document]
Comment (davidu1975) [reply export_academy/views.py:480]: done

File: export_academy/views.py (hunk 1 above)
Comment (davidu1975) [reply export_academy/views.py:483]: done

File: sso/mixins.py (hunk 2 above)
Comment (davidu1975) [reply sso/mixins.py:96]: yes it is

**Review by mforner13 — APPROVED** [review]
👍

**Pull request merged**